main.go

```diff
@@ -105,8 +105,7 @@ func main() {
 		os.Exit(1)
 	}
 
-	strg, err := tokenstorage.New(mgr.GetClient())
-
+	strg, err := tokenstorage.NewVaultStorage(mgr.GetClient(), "spi-controller-manager", cfg.VaultHost, cfg.ServiceAccountTokenFilePath)
```

Comment:
Is it ok to hardcode the `role` to `spi-controller-manager` here?

Reply:
This is not kubernetes role. it is vault "equivalent to serviceaccount" and it is set up here https://github.com/redhat-appstudio/service-provider-integration-operator/pull/88/files#diff-862cfcfe83b93917c4a2f5b5e7f4c00fa4cb7af432e447d0b8eb442ea4a00d43R130. It is saying that role `spi-controller-manager` (which is just a name and can be anything) expects kubernetes serviceaccount `spi-controller-manager` from `spi-system` namespaces and gives this role `spi` vault permissions (defined here https://github.com/redhat-appstudio/service-provider-integration-operator/pull/88/files#diff-862cfcfe83b93917c4a2f5b5e7f4c00fa4cb7af432e447d0b8eb442ea4a00d43R130)

Reply:
Ok, seems safe enough. If need be, we can make this configurable in the future to prevent clashes, but the name seems specific enough that I would not worry about it too much at this moment.